Storage/SQL/SQLiteHistory.swift

```diff
@@ -124,12 +124,16 @@ private let topSitesQuery = "SELECT * FROM \(TableCachedTopSites) ORDER BY frece
 extension SQLiteHistory: BrowserHistory {
     public func removeSiteFromTopSites(site: Site) -> Success {
         if let host = site.url.asURL?.normalizedHost() {
-            return db.run([("UPDATE \(TableDomains) set showOnTopSites = 0 WHERE domain = ?", [host])])
-                >>> { return self.refreshTopSitesCache() }
+            return self.removeHostFromTopSites(host)
         }
         return deferMaybe(DatabaseError(description: "Invalid url for site \(site.url)"))
     }
 
+    public func removeHostFromTopSites(host: String) -> Success {
```

Comment:
> I can convert both suggestedSites and Sites into TopSiteItems and compare them that way

Another approach to having the concrete type of `TopSiteItem` is to create a new protocol named `TopSite` and have extensions of both a `SuggestedSite` and `Site`. Then you wouldn't even have to convert it!

> sidenote: I was really hoping to just take the asset name of a SuggestedSite icon and set it to the faviconURL and have SDWebImage load it. I'd have to make 0 changes that way. SDWebImage does support loading files locally. But sadly not from xcassets

I wonder if we can use the GCDWebServer instance and serve the tile icons like we do our error pages. That way you'd be able to use urls 😁